### PR TITLE
Ni-PoRep Go/cgo additions

### PR DESCRIPTION
**Pull request by rvagg**
~Some additional changes on top of a rebased https://github.com/filecoin-project/filecoin-ffi/pull/453, while I try and wire things up for integration testing.~

~This has local linking for ref-fvm for https://github.com/filecoin-project/ref-fvm/pull/1999 on top of master for https://github.com/filecoin-project/ref-fvm/pull/2009.~

~It also has local linking for rust-filecoin-proofs-api for https://github.com/filecoin-project/rust-filecoin-proofs-api/pull/103~

--

Additions on top of https://github.com/filecoin-project/filecoin-ffi/pull/453, but this depends on https://github.com/filecoin-project/go-state-types/pull/269 being merged.

Note that his also includes `seal_commit_phase2_circuit_proofs` in `rust/src/proofs/api.rs`, that could go into #453.

### CODE REVIEW

File: proofs.go
Comment (rvagg) [inline proofs.go:422]: @vmx do you think I describe this properly here?

Comment (vmx) [reply proofs.go:422]: Yes, sounds good.

File: go.mod
Comment (Stebalien) [inline go.mod:8]: I'd cut a release if we're ready.

**Review by Stebalien — APPROVED** [review]
I can't comment on the proofs side of things, but:

1. The constants all look correct.
2. The FFI side of things looks correct.

File: go.mod
Comment (rvagg) [reply go.mod:8]: Agreed, it was just waiting for approval but I've given it a tick so @rjan90 can move forward there.

**Comment by rvagg** [comment]
Got something funky going on with the gocheck2 .. on aarch it wants it as a GOEXPERIMENT=gocheck2, but Darwin and Linux builders don't recognise that. I'll have to figure that one out before this can be landed.

**Comment by Stebalien** [comment]
Specifically:

- darwin/linux x86 expect the old form.
- linux aarch64 expect the GOEXPERIMENT version.

???

**Comment by Stebalien** [comment]
Ah: the aarch64 runner is "ubuntu-2204:current" while the others are `cimg/go:1.20`. That latter one likely needs to be upgraded to go 1.21.

**Comment by BigLep** [comment]
@rvagg : I know it's getting late your time.  Will you be handling the CI issues discussed above before signing off, or are you expecting @rjan90 to?

File: go.mod
Comment (rjan90) [reply go.mod]: Updated to go-state-types v0.14.0-rc1 in: https://github.com/filecoin-project/filecoin-ffi/pull/456/commits/ca6756413c0c8fded7b56d07b4fcc9a7c890afbb

**Comment by rjan90** [comment]
Thanks to @vmx he has addressed the CI-issues above, and also consolidated this PR into https://github.com/filecoin-project/filecoin-ffi/commit/9b6be8a39271c9761541b1f7bc7e1050e4f59f95

**Comment by BigLep** [comment]
I'm closing given superseded by https://github.com/filecoin-project/filecoin-ffi/pull/459

**Pull request closed without merging**
